pkg/edgeview/src/system.go

```diff
@@ -938,6 +967,217 @@ func dispAFile(f os.FileInfo) {
 	fmt.Printf("%s, %v, %d, %s\n", f.Mode().String(), f.ModTime(), f.Size(), f.Name())
 }
 
+func getTarFile(opt string) {
+	execCmd := strings.SplitN(opt, "tar/", 2)
+	if len(execCmd) != 2 {
+		fmt.Printf("tar needs a / and path input\n")
+		return
+	}
+	source := execCmd[1]
+	printColor(" - tar cmd: "+source, colorCYAN)
+	closePipe(true)
+
+	fi, err := os.Stat(source)
+	if err != nil {
+		fmt.Printf("find directory error: %v\n", err)
+		return
+	}
+
+	if !fi.IsDir() {
+		fmt.Printf("the path %s is not a directory\n", source)
+		return
+	}
+
+	ok := checkBlockedDirs(source)
+	if !ok {
+		fmt.Printf("the directory %s can not be tarred\n", source)
+		return
+	}
+
+	// check the directory size not to exceed the max
+	sizeBytes := du(source, fi)
+	if sizeBytes > tarMaxBytes {
+		fmt.Printf("directory size %d, exceeds maximum %d\n", sizeBytes, tarMaxBytes)
+		return
+	}
+	baseDir := filepath.Base(source)
+	targetDir := "/tmp/"
```

Comment:
I am not sure that storing (potentially) large files in `/tmp/` is a good idea. Can we move them to `/persist`?

Reply:
ok. we'll need to have a new 'rw' volume mount on that. Let me use '/persist/tmp'